### PR TITLE
[ML][Inference] adds new default_field_map field to trained models

### PR DESCRIPTION
**Pull request by benwtrent**
Adds a new `default_field_map` field to trained model config objects. 

This allows the model creator to supply field map if it knows that there should be some map for inference to work directly against the training data.

The use case internally is having analytics jobs supply a field mapping for multi-field fields. This allows us to use the model "out of the box" on data where we trained on `foo.keyword` but the `_source` only references `foo`.

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelConfig.java [hunk 1]
```diff
@@ -226,6 +235,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (licenseLevel != null) {
             builder.field(LICENSE_LEVEL.getPreferredName(), licenseLevel);
         }
+        if (defaultFieldMappings != null) {
```
Comment (davidkyle) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelConfig.java]:
```suggestion
        if (defaultFieldMappings != null && defaultFieldMappings.isEmpty() == false) {
```

I prefer not to write empty collections.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/inference/TrainedModelConfigTests.java [hunk 2]
```diff
@@ -52,7 +53,11 @@ public static TrainedModelConfig createTestTrainedModelConfig() {
             randomBoolean() ? null : TrainedModelInputTests.createRandomInput(),
             randomBoolean() ? null : randomNonNegativeLong(),
             randomBoolean() ? null : randomNonNegativeLong(),
-            randomBoolean() ? null : randomFrom("platinum", "basic"));
+            randomBoolean() ? null : randomFrom("platinum", "basic"),
+            randomBoolean() ? null :
+                Stream.generate(() -> randomAlphaOfLength(10))
+                    .limit(randomIntBetween(1, 10))
+                    .collect(Collectors.toMap(Function.identity(), (k) -> randomAlphaOfLength(10))));
```
Comment (davidkyle) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ml/inference/TrainedModelConfigTests.java]: 👍 neat I never thought of building a random map this way

File: docs/reference/ml/ml-shared.asciidoc [hunk 3]
```diff
@@ -1506,6 +1506,13 @@ The estimated number of operations to use the trained model.
 `license_level`:::
 (string)
 The license level of the trained model.
+
+`default_field_mappings` :::
+(object)
+A string to string mapping that contains the default field mappings to use
+when inferring against the model. Any field mapping described in the
```
Comment (davidkyle) [inline docs/reference/ml/ml-shared.asciidoc]: Should we add a line here about multi fields which is the primary reason for this change?

**Review by davidkyle — APPROVED** [review]
LGTM

File: docs/reference/ml/ml-shared.asciidoc (hunk 3 above)
Comment (benwtrent) [reply docs/reference/ml/ml-shared.asciidoc]: For sure!

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**